docs/essentials/gc/unmanaged/implementingdispose.md

```diff
@@ -1 +1,442 @@
+
+## Implementing the dispose pattern for a base class
+
+If you implement the dispose pattern for a base class, you must provide the following: 
+
+> **Important**
+>
+> You should implement this pattern for all base classes that implement `IDisposable` and are not `sealed`.
+
+* A `Dispose` implementation that calls the `Dispose(Boolean)` method. 
+
+* A `Dispose(Boolean)` method that performs the actual work of releasing resources. 
+
+* Either a class derived from `SafeHandle` that wraps your unmanaged resource (recommended), or an override to the `Object.Finalize` method. The `SafeHandle` class provides a finalizer that frees you from having to code one. 
+
+Here's the general pattern for implementing the dispose pattern for a base class that uses a safe handle. 
+
+```csharp
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+
+class BaseClass : IDisposable
+{
+   // Flag: Has Dispose already been called?
+   bool disposed = false;
+   // Instantiate a SafeHandle instance.
+   SafeHandle handle = new SafeFileHandle(IntPtr.Zero, true);
+
+   // Public implementation of Dispose pattern callable by consumers.
+   public void Dispose()
+   { 
+      Dispose(true);
+      GC.SuppressFinalize(this);           
+   }
+
+   // Protected implementation of Dispose pattern.
+   protected virtual void Dispose(bool disposing)
+   {
+      if (disposed)
+         return; 
+
+      if (disposing) {
+         handle.Dispose();
+         // Free any other managed objects here.
+         //
+      }
+
+      // Free any unmanaged objects here.
+      //
+      disposed = true;
+   }
+}
+```
+
+> **Note**
+>
+> The previous example uses a [SafeFileHandle]()http://dotnet.github.io/api/Microsoft.Win32.SafeHandles.SafeFileHandle.html object to illustrate the pattern; any object derived from `SafeHandle` could be used instead. Note that the example does not properly instantiate its `SafeFileHandle` object. 
+
+Here's the general pattern for implementing the dispose pattern for a base class that overrides `Object.Finalize`. 
+
+```csharp
+using System;
+
+class BaseClass : IDisposable
+{
+   // Flag: Has Dispose already been called?
+   bool disposed = false;
+
+   // Public implementation of Dispose pattern callable by consumers.
+   public void Dispose()
+   { 
+      Dispose(true);
+      GC.SuppressFinalize(this);           
+   }
+
+   // Protected implementation of Dispose pattern.
+   protected virtual void Dispose(bool disposing)
+   {
+      if (disposed)
+         return; 
+
+      if (disposing) {
+         // Free any other managed objects here.
+         //
+      }
+
+      // Free any unmanaged objects here.
+      //
+      disposed = true;
+   }
+
+   ~BaseClass()
+   {
+      Dispose(false);
+   }
+}
+```
```

Comment:
The example above actually contains a `SafeHandle` to dispose, this one doesn't contain any resource to release. Shouldn't the two examples be consistent in this regard?